### PR TITLE
ViewCollectionGroup should allow adding and removing items directly

### PR DESCRIPTION
**Pull request by justinvdm**

### CODE REVIEW

**Comment by justinvdm** [comment]
Depends on #370

File: go/base/static/js/src/components/structures.js [hunk 1]
```diff
@@ -141,14 +151,26 @@
...
-      lookup.keys().forEach(this.remove, this);
+      lookup.keys().forEach(this.onMemberRemove, this);
       this.members.remove(key);
       return lookup;
     }
   });
 
+  var idOfModel = function(obj) {
+    return obj.id
+      ? obj.id
```
Comment (hodgestar) [inline go/base/static/js/src/components/structures.js:165]: Should this also do the `_(obj).result('id')` dance?

Comment (hodgestar) [inline go/base/static/js/src/components/structures.js:165]: After discussion with @justinvdm this is fine -- model.id can never be a function.

File: go/base/static/js/src/components/structures.js [hunk 2]
```diff
@@ -317,6 +330,21 @@
   // A self-maintaining, 'flattened' lookup of the views in a group of view
   // collections.
   var ViewCollectionGroup = LookupGroup.extend({
+    idOfModel: idOfModel,
```
Comment (hodgestar) [inline go/base/static/js/src/components/structures.js]: is this reference to `idOfModel` ever used?

**Comment by hodgestar** [comment]
:+1: after resolving the question about `idOfModel` on the ViewCollectionGroup.

File: go/base/static/js/src/components/structures.js (hunk 2 above)
Comment (justinvdm) [inline go/base/static/js/src/components/structures.js]: I don't think it is, thanks, will remove it.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**